tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,648 @@
+
+        ClassDB::add_property("GDExample", PropertyInfo(Variant::FLOAT, "amplitude"), "set_amplitude", "get_amplitude");
+    }
+
+    void GDExample::GDExample() {
+        // initialize any variables here
+        time_passed = 0.0;
+        amplitude = 10.0;
+    }
+
+    void GDExample::_process(float delta) {
+        time_passed += delta;
+
+        Vector2 new_position = Vector2(
+            amplitude + (amplitude * sin(time_passed * 2.0)),
+            amplitude + (amplitude * cos(time_passed * 1.5))
+        );
+
+        set_position(new_position);
+    }
+
+    void GDExample::set_amplitude(const float p_amplitude) {
+        amplitude = p_amplitude;
+    }
+
+    float GDExample::get_amplitude() const {
+        return amplitude;
+    }
+
+Once you compile the module with these changes in place, you will see that a
+property has been added to our interface. You can now change this property and
+when you run your project, you will see that our Godot icon travels along a
+larger figure.
+
+Let's do the same but for the speed of our animation and use a setter and getter
+function. Our ``gdexample.h`` header file again only needs a few more lines of
+code:
+
+.. code-block:: C++
+
+    ...
+        float amplitude;
+        float speed;
+    ...
+        void _process(float delta) override;
+        void set_speed(float p_speed);
+        float get_speed();
+    ...
+
+This requires a few more changes to our ``gdexample.cpp`` file, again we're only
+showing the methods that have changed so don't remove anything we're omitting:
+
+.. code-block:: C++
+
+    void GDExample::_bind_methods() {
+        ClassDB::bind_method(D_METHOD("get_speed"), &GDExample::get_speed);
+        ClassDB::bind_method(D_METHOD("set_speed", "p_speed"), &GDExample::set_speed);
+	    ClassDB::add_property("GDExample", PropertyInfo(Variant::FLOAT, "speed"), "set_speed", "get_speed");
+    }
+
+    void GDExample::GDExample() {
+        // initialize any variables here
+        time_passed = 0.0;
+        amplitude = 10.0;
+        speed = 1.0;
+    }
+
+    void GDExample::_process(float delta) {
+        time_passed += speed * delta;
+
+        Vector2 new_position = Vector2(
+            amplitude + (amplitude * sin(time_passed * 2.0)),
+            amplitude + (amplitude * cos(time_passed * 1.5))
+        );
+
+        set_position(new_position);
+    }
+
+    void GDExample::set_speed(float p_speed) {
+        speed = p_speed;
+    }
+
+    float GDExample::get_speed() const {
+        return speed;
+    }
+
+Now when the project is compiled, we'll see another property called speed.
+Changing its value will make the animation go faster or slower.
+
+.. note::
+
+    For simplicity, we've left out the optional parameters in the
+    add_property() method call. These parameters are
+    ``hint``, ``hint_string``, ``usage`` and ``class_name``. These can be used to
+    further configure how properties are displayed and set on the Godot side.
+
```

Comment:
add_property() could use a tutorial by itself if it is not already available

Reply:
There could be a more detailed tutorial about properties but this PR focuses on having a minimum working GDExtension first.

Reply:
I don't know, seeing we do amplitude first, we could add additional values to speed to provide a range just to indicate one possible use case. We can then change this note to indicate there are more options.

That said, I'm all for having a separate page that purely handles properties and goes through the various things that are possible with these parameters as we really lack examples of each one.

Reply:
@Gallilus @BastiaanOlij 
Alright, I added a property hint for the `speed` property. This should do for a simple example. I might be able to go into further detail in a follow-up PR

Reply:
Looks good, its enough to peek someones interest and give atleast one example of how to use hints.